perma_web/perma/tasks.py

```diff
@@ -302,586 +704,331 @@ def warn_on_exception(message="Exception in block:", exception_type=Exception):
-
-    def sleep_unless_seconds_passed(seconds):
-        delta = time.time() - start_time
-        if delta < seconds:
-            wait = seconds - delta
-            print("Sleeping for {}s".format(wait))
-            time.sleep(wait)
-
-    # By domain, code to run after the target_url's page onload event.
-    # (Wrap in a lambda function to delay execution.)
-    special_domains = {
-        # Wait for splash page to auto redirect
-        "www.forbes.com": lambda: sleep_unless_seconds_passed(24),
-        "forbes.com": lambda: sleep_unless_seconds_passed(24)
-    }
-    post_load_function = special_domains.get(link.url_details.netloc)
-
-    try:
-
-        # create a request handler class that counts requests and responses
-        proxied_requests = []
-        proxied_responses = []
-        proxied_pairs = []
-        count_lock = threading.Lock()
-        class CountingRequestHandler(WarcProxyHandler):
-            def _proxy_request(self):
-
-                # make sure we don't capture anything in a banned IP range
-                if not url_in_allowed_ip_range(self.url):
-                    return
+        # Start warcprox process. Warcprox is a MITM proxy server and needs to be running
+        # before, during and after the headless browser.
+        #
+        # Start a headless browser to capture the supplied URL. Also take a screenshot if the URL is an HTML file.
+        #
+        # This whole function runs with the local dir set to a temp dir by run_in_tempdir().
+        # So we can use local paths for temp files, and they'll just disappear when the function exits.
+
+        # basic setup
+        link = capture_job.link
+        target_url = link.safe_url
+        browser = warcprox_controller = warcprox_thread = display = None
+        have_content = have_warc = have_html = False
+        thread_list = []
+        successful_favicon_urls = []
+
+        def inc_progress(step_count, step_description):
+            """
+                Update capture_job's progress
+
+                (defined inside capture function so that
+                step is initialized at 0 for each capture job)
+            """
+            if not hasattr(inc_progress, 'step'):
+                inc_progress.step = 0
+            inc_progress.step = int(inc_progress.step) + step_count
+            safe_save_fields(capture_job, step_count=inc_progress.step, step_description=step_description)
+            print "%s step %s: %s" % (link.guid, inc_progress.step, step_description)
+
+        # By domain, code to run after the target_url's page onload event.
+        # (Wrap in a lambda function to delay execution.)
+        special_domains = {
+            # Wait for splash page to auto redirect
+            "www.forbes.com": lambda: sleep_unless_seconds_passed(24, start_time),
+            "forbes.com": lambda: sleep_unless_seconds_passed(24, start_time)
+        }
+        post_load_function = special_domains.get(link.url_details.netloc)
```

Comment:
Ooh, interesting....... not sure. Maybe it should be both? For instance, if somebody perma'ed a tiny.cc link, content_url would be right. Are there any circumstances where the target_url would be normal, and the content_url would be something weird, like a cdn or....